modules/InstallBasicPackageFiles.cmake

```
@@ -676,7 +676,10 @@ ${_compatibility_vars}
   # CMake >= 3.15 already checks for CMAKE_EXPORT_PACKAGE_REGISTRY in `export(PACKAGE)` (cf.
   # cf. https://cmake.org/cmake/help/latest/policy/CMP0090.html), and we effectively back-port
   # this behavior to earlier versions.
-  if(CMAKE_EXPORT_PACKAGE_REGISTRY OR CMAKE_VERSION VERSION_GREATER_EQUAL 3.15)
+  # Note that even never CMake versions may apply old policy behaviors if the consuming project
+  # requires a lower version of CMake (e.g. `cmake_minimum_required(VERSION 3.14)`), so the
+  # check for `CMAKE_EXPORT_PACKAGE_REGISTRY` is necessary for CMake >= 3.15 as well.
+  if(CMAKE_EXPORT_PACKAGE_REGISTRY)
```

Why not using `CMAKE_MINIMUM_REQUIRED_VERSION` instead?

```suggestion
  if(CMAKE_EXPORT_PACKAGE_REGISTRY OR CMAKE_MINIMUM_REQUIRED_VERSION VERSION_GREATER_EQUAL 3.15)
```

---

According to https://github.com/robotology/ycm/issues/262#issuecomment-518521525, the reason motivating the version check in the first place was "that it will be easier to spot and remove it when we will stop supporting CMake < 3.15". But this doesn't apply here; we may require CMake ≥3.15 someday, but I suppose we won't ever require `CMAKE_MINIMUM_REQUIRED_VERSION VERSION_GREATER_EQUAL 3.15`. Given that the extra condition wouldn't change the semantics, and that it won't serve as a marker for future removal, I'd prefer to omit it.